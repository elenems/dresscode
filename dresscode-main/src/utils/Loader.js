import React from "react";
import LoaderSVG from "../assets/svg/loader.svg";
export default function Loader() {
  return (
    <div className="loader">
      <div className="loader-container">
        <img src={LoaderSVG} alt="Loading..." />
      </div>
    </div>
  );
}
