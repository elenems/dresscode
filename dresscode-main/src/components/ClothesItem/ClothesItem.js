import React, { Component } from "react";
import { withRouter, Link } from "react-router-dom";
import axios from "axios";
import Loader from "../../utils/Loader";
import HeartButton from "./HeartButton/HeartButton";
import Comments from "./Comments/Comments";
import WriteMessageContainer from "../WriteMessageContainer/WriteMessageContainer";
import PropTypes from "prop-types";

class ClothesItem extends Component {
  state = {
    clothesItem: null,
    isLoading: true,
    clothesUser: null,
    message: ""
  };
  componentDidMount() {
    window.scrollTo(0, 0);
    axios
      .get(`/getClothesItem?clothesId=${this.props.match.params.id}`)
      .then(data => {
        this.setState({
          clothesItem: data.data,
          isLoading: false
        });
        return data;
      })
      .then(data => {
        axios.get(`/getUser?userId=${data.data.addedBy}`).then(userData => {
          this.setState({
            clothesUser: userData.data
          });
        });
        axios.post("/incrementClothesView", {
          clothesId: this.state.clothesItem.id
        });
      })
      .catch(e => {
        this.setState({ isLoading: false });
      });
  }


  render() {
    return this.state.isLoading ? (
      <Loader />
    ) : (
      <div>
        <div className="clothes-item row">
          <div className="clothes-item-image">
            <img
              alt={this.state.clothesItem.title}
              src={this.state.clothesItem.image}
            />
          </div>

          <div className="clothes-item-description">
            <div className="clothes-item-description-top">
              <div>
                <h4>{this.state.clothesItem.title}</h4>
                <p>Brand: {this.state.clothesItem.brand}</p>
                <p>Size: {this.state.clothesItem.size}</p>
                <p>Condition: {this.state.clothesItem.condition}</p>
              </div>
              <div>
                <HeartButton
                  clothesId={this.state.clothesItem.id}
                  likesCount={this.state.clothesItem.likesCount}
                  user={this.props.user}
                />
              </div>
            </div>
            <div className="clothes-item-description-bottom">
              <p>
                <b>${this.state.clothesItem.price}</b>
              </p>
            </div>
            <div className="clothes-item-user-info">
              {this.state.clothesUser === null ? (
                <p>Can't load clothes' owner</p>
              ) : (
                <div>
                  <h6>Product owner</h6>
                  <div className="clothes-item-user-details">
                    <Link to={`/users/${this.state.clothesUser.id}`}>
                      {this.state.clothesUser.lastName}{" "}
                      {this.state.clothesUser.firstName}
                    </Link>
                    <span>Registered: {this.state.clothesUser.signUpDate}</span>
                  </div>
                  <WriteMessageContainer
                    addedBy={this.state.clothesItem.addedBy}
                    user={this.props.user}
                  />
                </div>
              )}
            </div>
            <div className="clothes-item-end">
              <h6>Description</h6>
              <p>{this.state.clothesItem.description}</p>
              <hr />
              <p>Post date: {this.state.clothesItem.date.slice(0, -4)}</p>
            </div>
          </div>
        </div>
        <Comments
          clothesTitle={this.state.clothesItem.title}
          clothesId={this.state.clothesItem.id}
          user={this.props.user}
          comments={this.state.clothesItem.comments}
        />
      </div>
    );
  }
}

ClothesItem.propTypes = {
  user: PropTypes.object.isRequired
};

export default withRouter(ClothesItem);
