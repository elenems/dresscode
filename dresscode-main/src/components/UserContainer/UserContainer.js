import React, { Component } from "react";
import axios from "axios";
import { withRouter } from "react-router-dom";
import Loader from "../../utils/Loader";
import ClothesItem from "../ItemsList/ListItem/ListItem";
import WriteMessageContainer from "../WriteMessageContainer/WriteMessageContainer";
import PropTypes from "prop-types";

export class UserContainer extends Component {
  state = {
    user: null,
    message: "",
    isLoading: true,
    userClothes: []
  };
  componentDidMount() {
    axios
      .get(`/getUser?userId=${this.props.match.params.id}`)
      .then(data => {
        this.setState({
          isLoading: false,
          user: data.data
        });
      })
      .catch(e => {
        this.setState({
          isLoading: false,
          message: "Can't load user info"
        });
      });
    axios
      .get(`/getUserClothes?userId=${this.props.match.params.id}`)
      .then(data => {
        this.setState({
          userClothes: data.data.clothes
        });
      })
      .catch(e => {
        this.setState({ message: "Can't load user's clothes" });
      });
  }
  render() {
    return this.state.isLoading ? (
      <Loader />
    ) : (
      <div className="row">
        {this.state.message ? (
          <p className="message">{this.state.message}</p>
        ) : (
          <div className="user-info">
            <h2>
              {this.state.user.firstName} {this.state.user.lastName}
            </h2>
            <p>{this.state.user.email}</p>
            <p>Location: {this.state.user.location}</p>
            <WriteMessageContainer
              addedBy={this.state.user.userId}
              user={this.props.user}
            />
          </div>
        )}

        <div className="user-clothes-container">
          <h4>All user items</h4>
          <div className="user-clothes-list">
            {!this.state.userClothes.length
              ? null
              : this.state.userClothes.map(item => {
                  return (
                    <ClothesItem
                      key={item.id}
                      user={this.props.user}
                      item={item}
                    />
                  );
                })}
          </div>
        </div>
      </div>
    );
  }
}

UserContainer.propTypes = {
  user: PropTypes.object.isRequired
};

export default withRouter(UserContainer);
