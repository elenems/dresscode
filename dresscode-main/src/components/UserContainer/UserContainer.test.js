import { shallow } from "enzyme";
import React from "react";
import { UserContainer } from "./UserContainer";
import Loader from "../../utils/Loader";
// ItemsList/ListItem/ListItem";
import ClothesItem from '../ItemsList/ListItem/ListItem';
describe("UserContainer", () => {
  const props = {
    user: {
      firstName: "Bill",
      lastName: "Johnson"
    },
    match: {
      params: {
        id: 1
      }
    }
  };
  it("Matches snapshot", () => {
    const wrapper = shallow(<UserContainer {...props} />).debug();
    expect(wrapper).toMatchSnapshot();
  });

  it("Show loader while laoding", () => {
    const wrapper = shallow(<UserContainer {...props} />);
    expect(wrapper.find(Loader).length).toBe(1);
  });

  it("Shows message", () => {
    const wrapper = shallow(<UserContainer {...props} />);
    wrapper.setState({ isLoading: false, message: "message" });
    expect(wrapper.find(".message").text()).toBe("message");
  });

  it("Shows user info if no message", () => {
    const wrapper = shallow(<UserContainer {...props} />);
    wrapper.setState({
      isLoading: false,
      user: { firstName: "bill", lastName: "Johnson", userId:"ids" }
    });
    expect(wrapper.find(".user-info").length).toBe(1);
  });

  it("Shows ClothesItem", ()=>{
    const wrapper = shallow(<UserContainer {...props} />);
    wrapper.setState({
        isLoading: false,
        user: { firstName: "bill", lastName: "Johnson", userId:"ids" },
        userClothes:[{id:1}]
      });
      expect(wrapper.find(ClothesItem).length).toBe(1)
  })
});
