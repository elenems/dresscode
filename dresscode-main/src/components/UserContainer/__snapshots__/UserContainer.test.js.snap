// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserContainer Matches snapshot 1`] = `"<Loader />"`;
